sci-misc/boinc/files/boinc.init

```diff
@@ -23,30 +23,45 @@ create_work_directory() {
 		chown "${USER}:${GROUP}" "${RUNTIMEDIR}"
 	fi
 
-	if [[ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]]; then
+	if [ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]; then
 		ln -s /etc/ssl/certs/ca-certificates.crt "${RUNTIMEDIR}"/ca-bundle.crt
 	fi
 
 	return 0
 }
 
 
+fix_lib_symlinks() {
+	local src="$1"
+	local tgt="$2"
+
+	if test -n "${tgt}"; then
+		# Remove a broken symlink
+		if test -h "${tgt}" \
+			-a "${src}" != "$(readlink "${tgt}")" ; then
+			rm -f "${tgt}"
+		fi
+
+		# symlink the correct path
+		if test -n "${src}" -a -f "${src}" -a ! -h "${tgt}"; then
```

Comment:
Wouldn't it be simpler to add `else; return` in the condition above, to cover the case where `tgt` exists and is valid?